### DISCUSSION TITLE
[Question]: Arzue OpenAI DALL-E-3 Plugin issue

### DISCUSSION THREAD

**Discussion by tensorstart** [danny-avila/LibreChat#2063] [Troubleshooting]
### What is your question?

RESPONSE FROM DALL-E-3：Failed to save the image locally. Failed to process the image with local. Cannot destructure property 'bytes' of '(intermediate value)' as it is null.
I wonder how to fix it. Thank you!

### More Details

![error](https://github.com/tensorstart/pandas_prec/blob/005b8c747a7f4ad4fd664fe70ce16d2fc2b3c45f/photos/2024-03-11_21h34_14.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by tensorstart** [comment]
![Screenshot_2024-03-11-22-27-54-26_320a9a695de7cdce83ed5281148d6f19.jpg](https://github.com/danny-avila/LibreChat/assets/40855931/02a25e84-69e4-416d-a8ec-2c1730d8c049)

add picture

**Reply by danny-avila** [reply]
This probably failed due to your configuration but this specific issue is going to be fixed here: https://github.com/danny-avila/LibreChat/pull/2064

**Reply by tensorstart** [reply]
@danny-avila Hello danny ,I used.env to deploy azure before, now I try to deploy librechat.yaml, but reported "No endpoint selected" 
My .env file set as follow:
```bash

DEBUG_LOGGING=true
DEBUG_CONSOLE=false

#=============#
# Permissions #
#=============#

# UID=1000
# GID=1000

#===============#
# Configuration #
#===============#
# Use an absolute path, a relative path, or a URL

CONFIG_PATH="/home/usrs/repos/LibreChat/librechat.yaml"

#===================================================#
#                     Endpoints                     #
#===================================================#


ENDPOINTS=azureOpenAI

PROXY=

```

and librechat.yaml file 
```bash
version: 1.0.4

# Cache settings: Set to true to enable caching
cache: true

# Example Azure OpenAI Object Structure

endpoints:
  azureOpenAI:
    # Endpoint-level configuration
    titleModel: "gpt-3.5-turbo"
    plugins: false
    assistants: true
    groups:
    # Group-level configuration
    - group: "rg_OpenAI_ChatGPT"
      apiKey: "fe6bxxxxxxxxx025"
      instanceName: "openaichatgpt"
      version: "2024-02-15-preview"
      # baseURL: "https://openaichatgpt.openai.azure.com/openai/deployments/gpt-3.5-turbo/chat/completions?api-version=2024-02-15-preview"
      # Model-level configuration
      models:
        gpt-4-vision-preview:
          deploymentName: gpt-4-vision-preview
          version: "2024-02-15-preview"
        gpt-3.5-turbo:
          deploymentName: gpt-35-turbo
        gpt-4-1106-preview:
          deploymentName: gpt-4-1106-preview
```
Is there anything I'm missing, thank you

**Reply by danny-avila** [reply]
DALL-E setup has not changed: https://docs.librechat.ai/install/configuration/azure_openai.html#generate-images-with-azure-openai-service-dall-e

You have `plugins` set to false, not sure if that's what you want.

**Reply by tensorstart** [reply]
Thanks for the reply, I'm trying to redeploy Azure OpenAI's chat feature with librechat.yml, but I'm getting an error of **No endpoint selected** with the configuration above

**Reply by danny-avila** [reply]
Try without `CONFIG_PATH` since it looks like you're using the default path, anyway. The server starts with no warnings/errors?

**Reply by tensorstart** [reply]
@
![IMG_0743](https://github.com/danny-avila/LibreChat/assets/40855931/b92d82ef-82cd-42ce-b37b-82a058c1da34)
The server works fine
I used docker-compose to setup your project, should I need to add librechat.yml path to docker-compose.yml and then rebuild docker compose?

volumes:
      - ./.env:/app/.env
      - ./images:/app/client/public/images
      - ./logs:/app/api/logs

**Reply by danny-avila** [reply]
Can I see your server logs?

>  I used docker-compose to setup your project, should I need to add librechat.yml path to docker-compose.yml and then rebuild docker compose?

Yeah, this is a necessary step except we recommend setting the volume mount in `docker-compose.override.yaml`: 


The following is from: https://docs.librechat.ai/install/configuration/custom_config.html#docker-setup

For Docker, you need to make use of an[ override file ](https://docs.librechat.ai/install/configuration/docker_override.html), named docker-compose.override.yml , to ensure the config file works for you.

- First, make sure your containers stop running with docker compose down
- Create or edit existing docker-compose.override.yml at the root of the project:

For more details on the override file, see the Docker Override Guide: https://docs.librechat.ai/install/configuration/docker_override.html

```yaml
version: '3.4'

services:
  api:
    volumes:
      - ./librechat.yaml:/app/librechat.yaml # OR local/filepath:container/filepath
```

Note: If you are using CONFIG_PATH for an alternative filepath for this file, make sure to specify it accordingly.

Start docker again, and you should see your config file settings apply

> docker compose up # no need to rebuild

I'll add a note in the Azure docs about the docker setup

**Reply by tensorstart** [reply]
yes, of course, I just create a docker-compose.override.yml at the root of the project but still repported No endpoint selected
 
![IMG_0744](https://github.com/danny-avila/LibreChat/assets/40855931/0de27578-5805-4da3-8cc7-f2e23651f2bf)

No error log and debug log as follow：
```
2024-03-16 14:29:16 info: [Optional] Redis not initialized. Note: Redis support is experimental.
2024-03-16 14:29:17 info: Connected to MongoDB
2024-03-16 14:29:17 debug: [indexSync] There are 143 messages in the database, 143 indexed
2024-03-16 14:29:17 debug: [indexSync] There are 34 convos in the database, 34 indexed
2024-03-16 14:29:17 info: Custom config file missing or YAML format invalid.
 
Check out the latest config file guide for configurable options and features.
https://docs.librech/... [truncated]
2024-03-16 14:29:17 info: Server listening on all interfaces at port 3080. Use http://localhost:3080/ to access it
2024-03-16 14:31:12 debug: [validateJson] files
["\"Ai_PDF.json\"","\"BrowserOp.json\"","\"Dr_Thoths_Tarot.json\"","\"DreamInterpreter.json\"","\"VoxScript.json\"","\"askyourpdf.json\"","\"drink_maestro.json\"","\"earthImagesAndVisualizations.json\"","\"image_prompt_enhancer.json\"","\"qrCodes.json\"","\"scholarai.json\"","\"uberchord.json\"","\"web_search.json\""]
 
```
Sorry to keep you waiting, my cat just chewed through my internet cable
